x-pack/solutions/security/plugins/security_solution/server/endpoint/services/scripts_library/scripts_library_client.ts

```diff
@@ -466,10 +466,43 @@ export class ScriptsLibraryClient implements ScriptsLibraryClientInterface {
   }
 
   public async delete(scriptId: string): Promise<void> {
-    throw new ScriptLibraryError('Not implemented', 501);
+    const scriptSo = await this.getScriptSavedObject(scriptId);
+
+    await this.soClient
+      .delete(SCRIPTS_LIBRARY_SAVED_OBJECT_TYPE, scriptId)
+      .catch(catchAndWrapError.withMessage(`Failed to delete script with id: ${scriptId}`));
+
+    try {
```

Comment:
I am wondering if an orphaned file could ever become a problem. Is there any background task that could potentially remove the orphaned files from time to time?